### ISSUE TITLE
Supposed conflict with ZeresPluginLibrary & FavoriteMedia Plugin

### ISSUE THREAD

**Issue by Infinitay** [zerebos/BDPluginLibrary#240]
# Versions
Discord: `stable 342408 (bd62b7c)`
BetterDiscord Core: `1.10.2`
ZeresPluginLibrary: `2.0.22`
FavoriteMedia: `1.10.4`

# Problem
Having both FavoriteMedia plugin and ZeresPluginLibrary enabled, when you visit a channel that has media such as an image on it, Discord will crash. According to the stack-trace, the issue is FavoriteMedia plugin. However, the author has said your plugin library is the cause (https://github.com/Dastan21/BDAddons/issues/274) and proceeded to lock the issue thread. Looking at the code myself, I don't see how there's a conflict, but I also don't know the BetterDiscord, BDFDB, or 0 libraries codebase.

# Stack-trace
```
web.<hash>.js:12 TypeError: Cannot read properties of undefined (reading 'Messages')
    at MediaFavButton.componentDidMount (FavoriteMedia.plugin.js:1017:141)
    at sX (web.<hash>.js:12:1028028)
    at e (web.<hash>.js:12:1048602)
    at e (web.<hash>.js:12:1048498)
    at web.<hash>.js:12:1048613
    at oB (web.<hash>.js:12:1049125)
    at oy (web.<hash>.js:12:1032474)
    at b (web.<hash>.js:12:1230240)
    at D (web.<hash>.js:12:1230770)
    at web.<hash>.js:1:752248
    at O (web.<hash>.js:1:752013)
    at MessagePort.L (web.<hash>.js:1:752072)
```

Respective FavoriteMedia.plugin.js code: https://github.com/Dastan21/BDAddons/blob/b88d167d59c7791e35e9ecc2eaab9118f33da2b9/plugins/FavoriteMedia/FavoriteMedia.plugin.js#L1017
```js
    componentDidMount () {
      this.tooltipFav = createTooltip(this.refs.tooltipFav, this.isFavorited ? Strings.Messages.GIF_TOOLTIP_REMOVE_FROM_FAVORITES : Strings.Messages.GIF_TOOLTIP_ADD_TO_FAVORITES, { style: 'primary' })
      Dispatcher.subscribe('FM_FAVORITE_MEDIA', this.updateFavorite)
    }
```

# Mitigation
Deleting ZeresPluginLibary or FavoriteMedia lets you use Discord again as per usual. I tried disabling ZeresPluginLibrary and CTRL+R, but I was still crashing. My issue only fixed when I renamed the plugin file to `0PluginLibrary.plugin.js.BAK` - or effectively deleting it.

I would appreciate it if you could take a deeper look into this and to confirm if the issue is with your plugin library and fix it if possible. It sucks having to choose between either using FavoriteMedia or plugins that rely on your plugin library because I use both effected plugins. In the mean time I guess I'll disable/remove FavoriteMedia until a solution or work-around is found. Thank you.

**Comment by AuriteCosmon** [comment]
I'd like to add this also happens when using the BetterChannelList plugin made by arg0NNY https://github.com/arg0NNY/DiscordPlugins/tree/cb0a3dd12c50163e1c70f22c8623887a44b31ed2/BetterChannelList
It'd be of great use to compile a list of all the plugins using the library which happen to have a conflict with it upon loading.

**Comment by bob-carter** [comment]
Since it might not be as clear as OP says, FavoriteMedia Plugin does use and require Zeres, so the issue being from this makes sense to some level, however if the issue is only from Fav's side it suggests some function used from said library is no longer working how it used to, or is deprecated, or said function is bugged.

**Comment by Infinitay** [comment]
> Since it might not be as clear as OP says, FavoriteMedia Plugin does use and require Zeres

Oh I'm sorry I wasn't aware of that. When I disabled Zere I did see a pop-up prompt appear now that you mention it asking to download ZerePluginLibrary, however **IIRC** I don't remember seeing the plugin description update to the typical message that reads something along the lines of _This plugin won't work correctly until you install ZerePluginLibrary_.

**Comment by bob-carter** [comment]
> > Since it might not be as clear as OP says, FavoriteMedia Plugin does use and require Zeres
> 
> Oh I'm sorry I wasn't aware of that. When I disabled Zere I did see a pop-up prompt appear now that you mention it asking to download ZerePluginLibrary, however **IIRC** I don't remember seeing the plugin description update to the typical message that reads something along the lines of _This plugin won't work correctly until you install ZerePluginLibrary_.

Yeah it's a known requirement, and is automatically downloaded when you enable it if I recall. The plugins code also says as much if you look at it.

**Comment by ISAKEM** [comment]
Downgrading FavoriteMedia doesnt work
Downgrading BDFDB doesnt work

**Comment by bob-carter** [comment]
> Downgrading FavoriteMedia doesnt work Downgrading BDFDB doesnt work

Likely because it tries to grab functions from a github file, since these just copy from internet hosted files to get their functions rather then having them all downloaded locally.

**Comment by Infinitay** [comment]
@ISAKEM I think FavoriteMedia was in fact the issue since there was no updates for Zeres library, but Dastan updated FavoriteMedia, and now it's working again for me. All I did was replace my FavoriteMedia.plugin.js with the current latest available on his repo (1.11.1 technically, but I think he forgot to update the version as the file has it under 1.11.0). Feel free to try the same too:

https://raw.githubusercontent.com/Dastan21/BDAddons/refs/heads/main/plugins/FavoriteMedia/FavoriteMedia.plugin.js

I'm not sure if he implemented a workaround and Zeres is still broken or not, but I guess the following commit specifically fixed this specific issue https://github.com/Dastan21/BDAddons/commit/9bdb55b6baf56f6058e0b265f7f8e73d9ffefc7b

**Comment by zerebos** [comment]
This plugin has reached EOL and will only be receiving high priority fixes!
